packages/core/utils/lib/convert-query-params.js

```diff
@@ -185,8 +186,31 @@ const convertPopulateObject = (populate, schema) => {
       return acc;
     }
 
-    // FIXME: This is a temporary solution for dynamic zones that should be
-    // fixed when we'll implement a more accurate way to query them
+    // Allow adding an 'on' strategy to populate queries for polymorphic relations, media and dynamic zones
+    const isAllowedAttributeForFragmentPopulate =
```

Comment:
we have some utils for content types in the utils like isRelation or similar you could add this there instead like 
isMorphAttribute ?